### PR TITLE
Don't validate provider ID in demo environment

### PR DESCRIPTION
**Pull request by spokenbird**

### CODE REVIEW

**Review by analoo — APPROVED** [review]

**Comment by analoo** [comment]
Would you please confirm that this configuration overrides that is in heroku once you've merged?

**Review by cram-cfa — APPROVED** [review]

**Review by enyia21 — CHANGES_REQUESTED** [review]
@spokenbird  We should set `allow-provider-registration-flow` to true so that Carl can do acceptance testing on Heroku branches.  I had this in my `test-main` PR but I did not merge it into main.  Carl used `test-main` for acceptance testing before it was closed.  `allow-provider-registration-flow` flag needs to be on to do acceptance testing on Heroku branches.
```suggestion
    validate-provider-id: ${VALIDATE_PROVIDER_ID_FLAG:false}
    allow-provider-registration-flow: ${ALLOW_PROVIDER_REGISTRATION_FLOW:true}
```

File: src/main/resources/application-demo.yaml [hunk 1]
```diff
@@ -13,3 +13,5 @@ form-flow:
       gcc:
         prefix: DEMO-
         creation-point: creation
+il-gcc:
+  validate-provider-id: false
```
Comment (enyia21) [inline src/main/resources/application-demo.yaml:17]: Carl used `test-main` for acceptance testing before it was closed.  `allow-provider-registration-flow` flag needs to be on to do acceptance testing on Heroku branches.

```suggestion
    validate-provider-id: ${VALIDATE_PROVIDER_ID_FLAG:false}
    allow-provider-registration-flow: ${ALLOW_PROVIDER_REGISTRATION_FLOW:true}
```

**Review by enyia21 — CHANGES_REQUESTED** [review]

**Comment by cram-cfa** [comment]
We should actually close this PR if my new PR is 👍 at https://github.com/codeforamerica/il-gcc-form-flow/pull/881

My PR will create fake providers for the demo environment only, so we can actually test for real!

**Review by cram-cfa — CHANGES_REQUESTED** [review]
Let's change this PR to *not* disable the provider Id validation, but to enable the flow per @enyia21 's comment! 

My PR will solve the data problem, and we should just prioritize getting that into demo if possible.

**Comment by spokenbird** [comment]
@cram-cfa @enyia21 I'm confused about what's being asked. Originally this was a quick fix to unblock testing in Heroku environments, but it sounds like Marc's #881 will solve that issue. 

Why do we want to set `allow-provider-registration-flow` here instead of just closing this PR in favor of Marc's change?

**Pull request closed without merging**